packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -0,0 +1,391 @@
+    return InheritedModel.inheritFrom<CupertinoTheme>(context)?.data?.copyWith(
+      brightness: brightness,
+      primaryColor: primaryColor,
+      primaryContrastingColor: primaryContrastingColor,
+      textTheme: textTheme,
+      barBackgroundColor: barBackgroundColor,
+      scaffoldBackgroundColor: scaffoldBackgroundColor,
+      tableBackgroundColor: tableBackgroundColor,
+    );
+  }
+}
+
+/// Styling specifications for a [CupertinoTheme].
+///
+/// All constructor parameters can be null, in which case a
+/// [CupertinoColors.activeBlue] based default iOS theme styling is used.
+///
+/// Parameters can also be partially specified, in which case some parameters
+/// will cascade down to other dependent parameters to create a cohesive
+/// visual effect. For instance, if a [primaryColor] is specified, it would
+/// cascade down to affect some fonts in [textTheme] if [textTheme] is not
+/// specified.
+///
+/// When retrieved using [CupertinoTheme.of], reading each property will create
+/// a dependency in the [BuildContext] given to [CupertinoTheme.of]. The
+/// given [BuildContext] will rebuild when that property changes.
+///
+/// See also:
+///
+///  * [CupertinoTheme], in which this [CupertinoThemeData] is inserted.
+///  * [ThemeData], a Material equivalent that also configures Cupertino
+///    styling via a [CupertinoThemeData] subclass [MaterialBasedCupertinoThemeData].
+@immutable
+class CupertinoThemeData extends Diagnosticable {
+  /// Create a [CupertinoTheme] styling specification.
+  ///
+  /// Unspecified parameters can be derived from another parameter or default
+  /// to an iOS default style.
+  const CupertinoThemeData({
+    Brightness brightness,
+    Color primaryColor,
+    Color primaryContrastingColor,
+    CupertinoTextTheme textTheme,
+    Color barBackgroundColor,
+    Color scaffoldBackgroundColor,
+    Color tableBackgroundColor,
+  }) : this.raw(
+        brightness,
+        primaryColor,
+        primaryContrastingColor,
+        textTheme,
+        barBackgroundColor,
+        scaffoldBackgroundColor,
+        tableBackgroundColor,
+      );
+
+  /// A raw constructor used by subclasses to populate the superclass's fields
+  /// to get superclass behaviors not overriden by subclasses.
+  ///
+  /// Same as the default constructor but with positional arguments to avoid
+  /// forgetting any.
+  @protected
+  const CupertinoThemeData.raw(
+    this._brightness,
+    this._primaryColor,
+    this._primaryContrastingColor,
+    this._textTheme,
+    this._barBackgroundColor,
+    this._scaffoldBackgroundColor,
+    this._tableBackgroundColor,
+  );
+
+  bool get _isLight => brightness == Brightness.light;
+
+  final Brightness _brightness;
+  /// The general brightness theme of the [CupertinoThemeData].
+  ///
+  /// Affects all other theme properties when unspecified. Defaults to
+  /// [Brightness.light].
+  ///
+  /// If coming from a Material [Theme] and unspecified, [brightness] will be
```

Comment:
Some subtle nuances here but it does not. CupertinoThemes under MaterialThemes are specific overrides. If you took that CupertinoThemeData and put it under a different MaterialTheme, it would use that new MaterialTheme's brightness. It wouldn't work if that CupertinoThemeData was constructed with the first material theme's brightness.

Reply:
OK, so it's:

By default, the MaterialTheme creates a CupertinoTheme that inherits the MaterialTheme's [ThemeData.brightness].